src/testdrive.F90

```diff
@@ -405,6 +539,34 @@ recursive subroutine run_unittest(test, unit, stat)
     !$omp critical(testdrive_testsuite)
     write(unit, '(a)') message
     !$omp end critical(testdrive_testsuite)
+    
+    stdout = ''
+    if (allocated(error)) then
+      stdout = trim(error%message)
+    endif
+    
+    s = index(message, '[')
+    e = index(message, ']')
+    res = message(s+1:e-1)
+    select case (res)
+      case ('FAILED', 'UNEXPECTED PASS')
+        call junitxml_write_testcase_opening_tag(test%name)
+        call junitxml_write_testcase_failure(res)
+        call junitxml_write_testcase_closing_tag(stdout)
+      case ('EXPECTED FAIL', 'PASSED')
+        if (len_trim(stdout) > 0) then  
+          call junitxml_write_testcase_opening_tag(test%name)
+          call junitxml_write_testcase_closing_tag(stdout)
+        else
+          call junitxml_write_testcase_single_tag(test%name)
```

Comment:
All these subroutines could be combined into a single one:
```fortran
call  junitxml_write_testcase(test%name, res, stdout)
```
with `res` and `stdout` being optional.

it could even be a function to be used as:
```
!$omp critical
select case(res)
 case(...)
  write(unit_junitxm, '(a)') junitxml_write_testcase(test%name, res, stdout)
 case(...)
end select
!$end omp critical
```
Such an approach (with a function returning a character string, instead of writing it directly to a file) could solve the issue raised by @awvwgk

Reply:
I am not sure about the dynamic size of the string variable. Please, refer to my response here: https://github.com/fortran-lang/test-drive/pull/27#discussion_r1315170193.
Please, let me know your thoughts.